### PR TITLE
Implemented CD for demos

### LINKED ISSUE #8: Setup CD for the demos
Goes hand in hand with vaxine-io/examples#9.
Code in this repository will actually deploy the demo, while the code in `vaxine-io/examples` will be used to trigger pipelines here

### PR DESCRIPTION
**Pull request by icehaunter**
This adds a special pipeline that is intended to be either triggered manually from the UI, or from another pipeline in BuildKite. It updates the target demo by changing the image in each deployed service to the one specified in the buildkite metadata, which can be set either with a block step, or on a job trigger.

Closes #8

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+all:
+
+update_demo:
+	terraform apply -var="$(target_demo)_image_tag=$(demo_image_tag)" -target=module.demo_apps.module.$(target_demo)
```
Comment (thruflo) [inline Makefile:4]: All looks good to me. The only thing I note is that in the past, I've kept terraform calls manual (and the resulting state checked in), to provision the resources. Then "pull and run latest container" can be done via a cloud build trigger.

I wonder, could terraform provision the trigger, and you keep the terraform calls manual? Or better this way?

Comment (icehaunter) [reply Makefile:4]: Resulting state may include sensitive info, so it's stored in a Cloud Storage bucket. That means you can run terraform commands locally or on CI and it'll be kept in sync. Triggers can be provisioned w/ Terraform, but I don't like using triggers when we also have this whole BuildKite shebang going on to be honest.

Anything you feel may go wrong with running scoped (`-target`) terraform here?

Comment (thruflo) [reply Makefile:4]: Yeah. Maybe I’m just being an old man with memories of terraform state going corrupt. Let’s run with what you have, given the scope of this terraform apply is limited.

**Comment by thruflo** [comment]
Merge away.

**Pull request merged**